src/types/account-id.ts

```diff
@@ -1,45 +1,55 @@
-import { makeClass } from "../utils/make-class";
-const { decodeAccountID, encodeAccountID } = require("ripple-address-codec");
-const { Hash160 } = require("./hash-160");
+import { decodeAccountID, encodeAccountID } from "ripple-address-codec";
+import { Hash160 } from "./hash-160";
 
-const AccountID = makeClass(
-  {
-    AccountID(bytes) {
-      Hash160.call(this, bytes);
-    },
-    inherits: Hash160,
-    statics: {
-      from(value) {
-        return value instanceof this
-          ? value
-          : /^r/.test(value)
-          ? this.fromBase58(value)
-          : new this(value);
-      },
-      cache: {},
-      fromCache(base58) {
-        let cached = this.cache[base58];
-        if (!cached) {
-          cached = this.cache[base58] = this.fromBase58(base58);
-        }
-        return cached;
-      },
-      fromBase58(value) {
-        const acc = new this(decodeAccountID(value));
-        acc._toBase58 = value;
-        return acc;
-      },
-    },
-    toJSON() {
-      return this.toBase58();
-    },
-    cached: {
-      toBase58() {
-        return encodeAccountID(this._bytes);
-      },
-    },
-  },
-  undefined
-);
+/**
+ * Class defining how to encode and decode an AccountID
+ */
+class AccountID extends Hash160 {
+  constructor(bytes: Buffer) {
+    super(bytes);
+  }
+
+  /**
+   * Defines how to construct an AccountID
+   *
+   * @param value either an existing AccountID, a hex-string, or a base58 r-Address
+   * @returns an AccountID object
+   */
+  static from(value: AccountID | string): AccountID {
+    return value instanceof this
```

Comment:
While terse, this is really confusing to read and will be more legible if you extract into two lines.

You can even comment it to make it clear what is happening.

```
// Handle AccountID parameters
if (value instanceof this) {
  return value
}

// Handle string parameters.
return /^r/.test(value) ? this.fromBase58Value : new AccountID(Buffer.from(value, "hex")
```